core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -42,25 +43,18 @@
  * A component to serve http requests, backed by rest handlers.
  */
 public class HttpServer extends AbstractLifecycleComponent<HttpServer> implements HttpServerAdapter {
-
-    private final Environment environment;
-
     private final HttpServerTransport transport;
 
     private final RestController restController;
 
     private final NodeService nodeService;
 
     @Inject
-    public HttpServer(Settings settings, Environment environment, HttpServerTransport transport,
-                      RestController restController,
-                      NodeService nodeService) {
+    public HttpServer(Settings settings, HttpServerTransport transport, RestController restController, NodeService nodeService) {
         super(settings);
-        this.environment = environment;
         this.transport = transport;
         this.restController = restController;
         this.nodeService = nodeService;
-        nodeService.setHttpServer(this);
```

Comment:
I see and I understand the reasoning. However, Guice circular dependency resolution is very tricky and if Guice can't resolve things correctly you get the most horrific, cryptic, MB long errors. This is why we are slowing removing Guice usage. I suspect there was a good reason as to why this is done this way and I rather not change this as part of this PR. Can we do that as part of the follow up too?